src/de/mod.rs

```diff
@@ -1054,26 +1054,20 @@ mod tests {
             assert_eq!(de.read, vec![]);
             assert_eq!(de.write, vec![]);
 
-            assert_eq!(
-                de.next().unwrap(),
-                Start(BytesStart::borrowed_name(b"root"))
-            );
-            assert_eq!(
-                de.peek().unwrap(),
-                &Start(BytesStart::borrowed_name(b"inner"))
-            );
+            assert_eq!(de.next().unwrap(), Start(QName(b"root").into()));
+            assert_eq!(de.peek().unwrap(), &Start(QName(b"inner").into()));
 
             // Should skip first <inner> tree
             de.skip().unwrap();
             assert_eq!(de.read, vec![]);
             assert_eq!(
                 de.write,
                 vec![
-                    Start(BytesStart::borrowed_name(b"inner")),
+                    Start(QName(b"inner").into()),
                     Text(BytesText::from_escaped_str("text")),
-                    Start(BytesStart::borrowed_name(b"inner")),
-                    End(BytesEnd::borrowed(b"inner")),
-                    End(BytesEnd::borrowed(b"inner")),
+                    Start(QName(b"inner").into()),
```

Comment:
I'm mildly negative about this because it makes using the API less consistent. That's not to say that the implementation shouldn't be there, because it makes sense to have the implementation, but in practice when used like this it flows more awkwardly when some tags use `QName` and others use `BytesText::from_*`, `BytesCdata::from_*`, etc.

Not a blocker, just FWIW

Reply:
I've made this change in order to reduce diff in one of following commits, that splits common `Event` into `reader::Event` and `writer::Event`. `reader::Event` will not have explicit public methods to construct a value (this is unnecessary for the user, if user need an event he could use `writer::Event`, reader events is read-only), so it will impossible to construct it in the doctests. I can move that change there